From: Peter Kokot <peterkokot@gmail.com>
Subject: Remove sed from php-config script

The Autoconf configure options are replaced in main/build-defs.h.in and
scripts/php-config.in. Instead of doing redundant quotes removal step
each time the php-config script is run by the user on PHP installation,
they can be removed before the variable substitution step directly only
at the PHP configuration build phase. This also removes the unused
CONFIGURE_OPTIONS Make variable in root and phpize generated Makefiles.
The configure options in main/build-defs.h.in remain quoted.

Also, sed substituted value can have issues in cases where build is
happening on one machine and targeted machine has different path to sed
command. For example, on build machine it is in /usr/bin/sed, and on
target in /bin/sed.

This should be further improved because some PHP installations manually
adjust the php-config script and inject also variables such as CFLAGS
and similar. For example, variables passed to configure after quotes
removal can include malformed value with spaces and without any quotes.
---
 build/php.m4          | 4 +++-
 scripts/php-config.in | 4 ----
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/build/php.m4 b/build/php.m4
index 933ada51dc..4909484a3b 100644
--- a/build/php.m4
+++ b/build/php.m4
@@ -2159,7 +2159,9 @@ EOF
   chmod +x $1
   CONFIGURE_COMMAND="$CONFIGURE_COMMAND $CONFIGURE_OPTIONS"
   PHP_SUBST_OLD(CONFIGURE_COMMAND)
-  PHP_SUBST_OLD(CONFIGURE_OPTIONS)
+  dnl Remove quotes.
+  CONFIGURE_OPTIONS=$(echo $CONFIGURE_OPTIONS | $SED -e "s#'##g")
+  AC_SUBST([CONFIGURE_OPTIONS])
 ])
 
 dnl
diff --git a/scripts/php-config.in b/scripts/php-config.in
index 9271e87286..8d82a240cc 100644
--- a/scripts/php-config.in
+++ b/scripts/php-config.in
@@ -1,6 +1,5 @@
 #! /bin/sh
 
-SED="@SED@"
 prefix="@prefix@"
 datarootdir="@datarootdir@"
 exec_prefix="@exec_prefix@"
@@ -41,9 +40,6 @@ else
   php_binary="$php_cgi_binary"
 fi
 
-# Remove quotes
-configure_options=`echo $configure_options | $SED -e "s#'##g"`
-
 case "$1" in
 --prefix)
   echo $prefix;;
